uc-crux-llvm/README.md

```diff
@@ -0,0 +1,267 @@
+[Crux] Attempting to prove verification conditions.
+[Crux] Attempting to prove verification conditions.
+[Crux] Results for double_free
+[Crux] Found likely bugs:
+Pointer freed twice
+```
+That's not too impressive, a simple linter might be able to find that bug. However, since `uc-crux-llvm` uses symbolic execution, it can precisely conclude that the following program _does not_ have a potential double-free (or _any_ other undefined behavior), provided that it's passed a non-null pointer:
+```c
+#include <stdlib.h>
+void not_double_free(int *ptr, int x) {
+  if (x % 2 == 0) {
+    printf("even!\n"); // needed to prevent Clang from optimizing away the if/else
+    free(ptr);
+  }
+  if ((x + 1) % 2 == 0) {
+    free(ptr);
+  }
+}
+```
+```
+$ uc-crux-llvm --entry-points not_double_free not_double_free.c
+[CLANG] clang "-c" "-DCRUCIBLE" "-emit-llvm" "-g" "-I" "test/programs" "-I" "/home/langston/code/crucible/crux-llvm/./c-src/includes" "-O1" "-o" "crux-build/not_double_free.bc" "test/programs/not_double_free.c"
+even!
+[Crux] Attempting to prove verification conditions.
+even!
+[Crux] Results for not_double_free
+[Crux] Function is safe if deduced preconditions are met:
+Arguments:
+  A pointer to uninitialized space for 1 elements: 
+  An integer:
+```
+While the examples here have very simple inputs, `uc-crux-llvm` is capable of synthesizing much richer inputs, including nested and recursive structs, pointers, floats, and more.
+
+## How It Works
+
+![Architecture of UC-Crux-LLVM](doc/uc-crux-llvm.svg)
+
+Once a target function has been selected (either by the user or in [Exploratory
+Mode](#Exploratory-Mode)), `uc-crux-llvm` will generate the smallest possible
+fully symbolic input for that function based on its type signature.
+`uc-crux-llvm` will symbolically execute the function on this input, and
+iteratively expand it:
+
+- If the symbolic execution succeeded with no safety condition violations,
+  it will increase the input depth (up to the user-specified bound) and execute
+  it again.
+
+- If some safety conditions were (potentially) violated, the `uc-crux-llvm` will
+  examine the error, and either:
+
+  - report it (and optionally continue executing to find other problems), or
+  - use heuristics to decide that the error was likely a false positive (i.e.,
+    due to a unsatisfied precondition, such as an uninitialized global
+    variable). The tool may then use further heuristics to generate inputs or a
+    program state/memory layout that satisfies the precondition, such as
+
+    - Allocating memory to back previously unmapped pointers in arguments
+    - Expanding the sizes of allocations
+    - Initializing global variables
+
+This approach is fairly different from UC-KLEE, which uses "lazy
+initialization", i.e., allocating memory *as it's used* by the program.
+
+## Building
+
+### Prerequisites
+
+Before running `uc-crux-llvm`, you'll need to install the following software:
+
+* GHC and `cabal`, preferably using `ghcup`:
```

Comment:
Maybe "we recommend" instead of "preferably", since there's no uc-crux-llvm advantage to using ghcup-provided GHC and cabal, it's just easy for the user.  To put it another way, if they already have GHC and cabal via other mechanisms, uc-crux-llvm should work fine.

Reply:
This section is copy/pasted nearly verbatim from the `crux-llvm` README, so if we want to change it we might want to defer it to another PR and tackle both, rather than include `crux-llvm` documentation changes in this one.

Reply:
I agree with @kquick about the wording here, and it's probably worth updating both.

Reply:
https://github.com/GaloisInc/crucible/issues/694